lib/querystring.js

```
@@ -45,8 +44,13 @@ exports.parse = function(str){
   return String(str)
     .split('&')
     .reduce(function(ret, pair){
-      var pair = decodeURIComponent(pair.replace(/\+/g, ' '))
-        , eql = pair.indexOf('=')
+      try{ 
+        var pair = decodeURIComponent(pair.replace(/\+/g, ' '))
+      } catch(e) { 
```

the resulting failure causes `pair` not short-circuit so there's no need for this, the following is fine:

``` js
      try{ 
        pair = decodeURIComponent(pair.replace(/\+/g, ' '));
      } catch(e) {
        // ignore
      }
```

though I hate to just ignore errors